### PR TITLE
add-on install for reviewers

### PR DESCRIPTION
**Pull request by ngokevin**
still testing it out a bit

screen shot 2015-09-28 at 4 55 14 pm

### CODE REVIEW

File: src/media/js/addon/actions/addon.js [hunk 1]
```diff
@@ -71,3 +80,28 @@ export function fetchVersions(addonSlug) {
...
+    Install add-on.
+  */
+  return (dispatch, getState) => {
+    const apiArgs = getState().apiArgs || {};
+    manifestUrl = Url(manifestUrl).q(apiArgs);
+
+    const req = window.navigator.mozApps.installPackage(manifestUrl);
+    dispatch(installBegin(addonSlug));
+
+    req.onsuccess= err => {
```
Comment (chuckharmston) [inline src/media/js/addon/actions/addon.js]: Missing space

File: src/media/js/addon/actions/addon.js [hunk 2]
```diff
@@ -71,3 +80,28 @@ export function fetchVersions(addonSlug) {
...
 }
+
+
+export function install(addonSlug, manifestUrl) {
+  /*
+    Install add-on.
+  */
+  return (dispatch, getState) => {
+    const apiArgs = getState().apiArgs || {};
+    manifestUrl = Url(manifestUrl).q(apiArgs);
```
Comment (chuckharmston) [inline src/media/js/addon/actions/addon.js]: Would be super cool if we could combine this pattern into a single function, we do it all over the place.

File: src/media/js/addon/components/install.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    return (
+      <div>
+        <h2>Install Add-on</h2>
+
+        <button disabled={this.props.isInstalled || this.props.isInstalling}
+                onClick={this.install}>
+          {this.getButtonText()}
+        </button>
+
+        {this.state.hasTriedInstall && this.props.installErrorMessage &&
```
Comment (chuckharmston) [inline src/media/js/addon/components/install.js]: Add a class around this and make it red?

File: src/media/js/addon/containers/reviewDetail.js [hunk 4]
```diff
@@ -31,15 +33,26 @@ export class AddonReviewDetail extends React.Component {
   }
 
   render() {
-    if (!this.props.addon || !this.props.addon.slug) {
+    const addon = this.props.addon;
+
+    if (!addon || !addon.slug) {
```
Comment (chuckharmston) [inline src/media/js/addon/containers/reviewDetail.js]: This makes perfect sense, but it's a really awkward pattern. There's gotta be a better way to do this. Hm.

File: src/media/js/addon/reducers/addon.js [hunk 5]
```diff
@@ -58,6 +58,44 @@ export default function addonReducer(state=initialState, action) {
       return newState;
     }
 
+    case addonActions.INSTALL_BEGIN: {
+      /*
+        Add-on installation begin.
+
+        payload (string) -- slug of add-on being installed.
+      */
```
Comment (chuckharmston) [inline src/media/js/addon/reducers/addon.js]: :+1:

File: src/media/js/addon/components/install.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  getButtonText() {
+    if (this.props.isInstalling) {
+      return 'Installing...';
+    } else if (this.props.isInstalled) {
+      return 'Installed';
+    } else {
+      return 'Install';
+    }
+  }
```
Comment (chuckharmston) [inline src/media/js/addon/components/install.js]: React is so beautiful. :joy:

**Comment by chuckharmston** [comment]
r+

File: src/media/js/addon/components/install.js (hunk 3 above)
Comment (ngokevin) [inline src/media/js/addon/components/install.js]: i'll just let you do this when you sweep the page

File: src/media/js/addon/containers/reviewDetail.js (hunk 4 above)
Comment (ngokevin) [inline src/media/js/addon/containers/reviewDetail.js]: besides moving to a diff render method, figuring out a pattern for fragment loading placeholders will be more involved

File: src/media/js/addon/actions/addon.js (hunk 2 above)
Comment (ngokevin) [inline src/media/js/addon/actions/addon.js]: maybe. gonna punt it

**Pull request merged**